### PR TITLE
Add tile module + BinaryFillHoles filter

### PR DESCRIPTION
**Pull request by alessiamarcolini**
* Add tile module
* Add BinaryFillHoles morphological filter, needed to check if the tile has enough tissue.

*Note for reviewers: it may be convenient to look only at the final diff instead of the diff for single commits.*

### CODE REVIEW

File: src/histolab/tile.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            Whether the image has enough tissue, i.e. if the proportion of tissue
+            over the total area of the image is more than ``threshold`` and the image
+            variance after morphological operations is more than
+            ``near_zero_var_threshold``.
+        """
+
+        if self._is_almost_white:
+            return False
+
+        if self._is_all_tissue_or_no_tissue(near_zero_var_threshold):
```
Comment (nicolebussola) [inline src/histolab/tile.py:57]: I do not remember why "all tissue" is a problem

Comment (alessiamarcolini) [reply src/histolab/tile.py:58]: It's not actually a problem. The point is: if we check for "no tissue" using the variance of the image, "no tissue" is the same as "all tissue". We probably need to find another way to determine if we have no tissue in the tile, then.

File: src/histolab/tile.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from .util import lazyproperty
+
+
+class Tile:
+    def __init__(self, image: Image.Image, coords: CoordinatePair, level: int = 0):
+        self._image = image
+        self._level = level
+        self._coords = coords
+
+    @lazyproperty
```
Comment (ernestoarbitrio) [inline src/histolab/tile.py:20]: why this property to return an attribute of the class? We do already have self._image without this prop

File: src/histolab/tile.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    def image(self) -> Image.Image:
+        return self._image
+
+    @lazyproperty
+    def level(self) -> int:
```
Comment (ernestoarbitrio) [inline src/histolab/tile.py:25]: also here

File: src/histolab/tile.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        return self._level
+
+    @lazyproperty
+    def coords(self) -> CoordinatePair:
```
Comment (ernestoarbitrio) [inline src/histolab/tile.py:29]: and here

File: src/histolab/tile.py (hunk 1 above)
Comment (ernestoarbitrio) [reply src/histolab/tile.py:58]: IMHO this prop name is odd, considering that it returns a bool, can we call it just `_is_all_tissue` ?

File: src/histolab/tile.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+        if self._is_all_tissue_or_no_tissue(near_zero_var_threshold):
+            return False
+
+        if not self._has_tissue_more_than_percent(tissue_percent):
+            return False
+
+        return True
+
+    def save(self, path):
```
Comment (ernestoarbitrio) [inline src/histolab/tile.py:66]: python typing missed

File: src/histolab/tile.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+from PIL import Image
+
+from .filters import image_filters as imf
+from .filters import morphological_filters as mof
+from .filters.util import mask_percent
+from .types import CoordinatePair
+from .util import lazyproperty
+
+
+class Tile:
```
Comment (ernestoarbitrio) [inline src/histolab/tile.py:14]: where's the docstring? XD

File: src/histolab/tile.py (hunk 1 above)
Comment (alessiamarcolini) [reply src/histolab/tile.py:58]: The point is that this evaluates to True both for cases of ALL tissue and cases of NO tissue at all. Mmmmm

File: src/histolab/tile.py (hunk 2 above)
Comment (alessiamarcolini) [reply src/histolab/tile.py:20]: Yes but is a way to expose `image` as a read-only attribute, right?

File: src/histolab/tile.py (hunk 6 above)
Comment (alessiamarcolini) [reply src/histolab/tile.py:15]: totally missed :D

File: src/histolab/tile.py (hunk 1 above)
Comment (ernestoarbitrio) [reply src/histolab/tile.py:69]: yes but it's a bool and the name can be more simple than that, can we try to change that name? @nicolebussola what do you think about it?

File: src/histolab/tile.py (hunk 2 above)
Comment (ernestoarbitrio) [reply src/histolab/tile.py:31]: oh ok i didn't realize we wanna have that as a read-only attr

Comment (alessiamarcolini) [reply src/histolab/tile.py:31]: yeah, Tile is a read-only container to me

**Pull request merged**